src/functions/slack/bot_commands.go

```diff
@@ -28,36 +28,21 @@ type SettleCommand struct {
 type HelpCommand struct {
 }
 
-
 func ParseBuyin(user string, text string) *BuyinCommand {
-	re := regexp.MustCompile(`in £?(\d+)(\.\d+)?`)
+	re := regexp.MustCompile(`in £?(\d+[.]?\d{1,2})?`)
 	m := re.FindStringSubmatch(text)
-
-	if m != nil {
-		pounds, err := strconv.Atoi(m[1])
-
-		if err != nil {
-			log.Printf("Couldn't parse value in buyin command: %v", err);
-			return nil
-		}
-
-		value := pounds * 100
-
-		if len(m) > 1 && len(m[2]) > 0 {
-			pence, err := strconv.Atoi(m[2][1:])
-
-			if err != nil {
-				log.Printf("Couldn't parse value in buyin command: %v", err);
-				return nil
-			}
-
-			value += pence
-		}
-
-		return &BuyinCommand{user, value}
+	if m == nil {
+		return nil
+	}
+	parsed, err := strconv.ParseFloat(m[1], 64)
```

Comment:
I think this will lose precision if you buy in or cash out for like £1B or so. Could you add a test for that please?

Reply:
You're right.. `1000000000.01` does not retain precision.

🤷‍♂ 

Also, parsing of single digits is broken cause the regex is bad.
Regex should be changed to something like: `in £?(\d+[.]?\d{0,2})?` or `in £?(\d*[.]?\d{1,2})?`

Reply:
instead we could do something like:
`in £?(\d*)[.]?(\d{0,2})?`

then parse m[1] as int and m[2] as float, so you only ever need 2dp of float precision